src/sidebar/store/modules/drafts.js

```diff
@@ -178,13 +180,29 @@ function getDraftIfNotEmpty(state, annotation) {
 /**
  * Returns a list of draft annotations which have no id.
  *
- * @return {Object[]}
+ * @return {Annotation[]}
  */
 const unsavedAnnotations = createSelector(
   state => state,
   drafts => drafts.filter(d => !d.annotation.id).map(d => d.annotation)
 );
 
+/**
+ * @typedef DraftsStore
+ *
+ * // Actions
+ * @prop {typeof createDraft} createDraft
+ * @prop {typeof deleteNewAndEmptyDrafts} deleteNewAndEmptyDrafts
+ * @prop {typeof discardAllDrafts} discardAllDrafts
+ * @prop {typeof removeDraft} removeDraft
+ *
+ * // Selectors
+ * @prop {() => boolean} countDrafts
+ * @prop {(a: Annotation) => Draft|null} getDraft
+ * @prop {(a: Annotation) => Draft|null} getDraftIfNotEmpty
+ * @prop {() => Annotation[]} unsavedAnnotations
```

Comment:
It is potentially problematic that drafts hold references to `Annotation`s as those copies may not be in sync with the `Annotation` in the `annotations` module. Fixing that is out of scope for this PR though.

Reply:
I'm not sure I fully understand the danger here. It appears it's actually return `Draft[]]`, perhaps cast to an `Annotation`, but it's returning data from the drafts store nevertheless. Are you implying there is a linkage between copies in the drafts store and annotations store that could potentially get out of sync? If that's the case we may need to redesign the drafts/annotations stores and set up a different type of relationship.

For now, I'm going to go ahead and change this type to {Draft[]}, we can discuss further on Monday.